packages/e2e-test-utils-playwright/src/metrics/index.ts

```diff
@@ -108,4 +146,132 @@ export class Metrics {
 				} )
 		);
 	}
+
+	/**
+	 * Returns the loading durations using the Navigation Timing API. All the
+	 * durations exclude the server response time.
+	 */
+	async getLoadingDurations(): Promise< LoadingDurations > {
+		return await this.page.evaluate( () => {
+			const [
+				{
+					requestStart,
+					responseStart,
+					responseEnd,
+					domContentLoadedEventEnd,
+					loadEventEnd,
+				},
+			] = performance.getEntriesByType(
+				'navigation'
+			) as PerformanceNavigationTiming[];
+			const paintTimings = performance.getEntriesByType(
+				'paint'
+			) as PerformancePaintTiming[];
+
+			const firstPaintStartTime = paintTimings.find(
+				( { name } ) => name === 'first-paint'
+			)?.startTime as number;
```

Comment:
Nit: Instead of casting as number, how about we just use `!`?

```suggestion
			)!.startTime;
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/53808/commits/feb23d53a20f9b7986339d636ba323d90c15e442